bindings/ruby/tests/test_buffered_port.rb

```diff
@@ -0,0 +1,41 @@
+gem 'test-unit'
+require 'test/unit'
+require 'test/unit/ui/console/testrunner'
+require 'yarp'
+
+class BufferdPortTest < Test::Unit::TestCase  
+  def open_and_connect()
+    p = Yarp::BufferedPortBottle.new()
+    p2 = Yarp::BufferedPortBottle.new()
+    assert_equal(true, p.open("/ruby/out"))
+    assert_equal(true, p2.open("/ruby/in"))
+    assert_equal(true,Yarp::Network::connect(p.getName(), p2.getName()))
+    p.close()
+    p2.close()
+  end
+  def open_and_connect_comm()
+    p = Yarp::BufferedPortBottle.new()
+    p2 = Yarp::BufferedPortBottle.new()
+    bt_out = p.prepare()
+    bt_out.addInt32(10)
+    assert_equal(true, p.open("/ruby/out"))
+    assert_equal(true, p2.open("/ruby/in"))
+    assert_equal(true,Yarp::Network::connect(p.getName(), p2.getName()))
+    p.write()
+    Yarp::delay(1.0)
+    bt_in=p2.read()
+    assert_equal(1, bt_in.size())
+    assert_equal(10, bt_in.get(0).asInt32())
+    p.close()
+    p2.close()
+  end
+end
+
+Yarp::Network.init()
```

Comment:
A `Yarp::Network.setLocalMode(true)` line here (not sure about Ruby syntax) would avoid warnings.